crates/kubelet/src/node.rs

```diff
@@ -4,8 +4,34 @@ use k8s_openapi::api::coordination::v1::Lease;
 use k8s_openapi::api::core::v1::Node;
 use k8s_openapi::apimachinery::pkg::apis::meta::v1::Time;
 use kube::api::{Api, DeleteParams, PatchParams, PostParams};
+use kube::Error;
 use log::{debug, error, info};
 
+macro_rules! retry {
+    ($action:expr, times: $num_times:expr, $on_err:expr) => {{
+        let mut n = 0u8;
+        let mut duration = std::time::Duration::from_millis(100);
+        loop {
+            n += 1;
+            let result = $action;
+            match result {
+                Ok(_) => break result,
+                Err(ref e) => {
+                    $on_err(e);
+                    tokio::time::delay_for(duration).await;
+                    duration *= (n + 1) as u32;
```

Comment:
Nit: Maybe we should just stick to a linear retry and then refactor later to use something like an exponential backoff with jitter because this feels like it is weirdly in between

Reply:
Why is linear retry preferred to exponential without jitter?

Reply:
I just realized this was a `*=` and not a `+=` so never mind me over here